content/client-server-api/modules/end_to_end_encryption.md

```diff
@@ -671,10 +671,7 @@ The process between Alice and Bob verifying each other would be:
 15. Assuming they match, Alice and Bob's devices calculate the HMAC of
     their own device keys and a comma-separated sorted list of the key
     IDs that they wish the other user to verify, using SHA-256 as the
-    hash function. HMAC is defined in [RFC
-    2104](https://tools.ietf.org/html/rfc2104). The key for the HMAC is
-    different for each item and is calculated by generating 32 bytes
-    (256 bits) using [the key verification HKDF](#hkdf-calculation).
+    hash function. The HMAC calculation is defined [below](#mac-calculation).
```

Comment:
I find it confusing the way we switch freely between "HMAC" and "MAC", and as far as I can tell we're talking about the same thing.

Also, I think we're calculating two separate MACs here? The current phrasing suggests that all the inputs are smashed together to calculate a single MAC.

I don't think we need to say we use SHA-256 here: that's part of the MAC calculation.

I also don't think we need to specify the format of the key IDs list; again that can go below.

```suggestion
15. Assuming they match, Alice and Bob's devices each calculate Message
    Authentication Codes (MACs) for:
    * their own device key [XXX: ed25519 key? curve25519 key? something else?], and:
    * the list of key IDs that they wish the other user to verify.
    The MAC calculation is defined [below](#mac-calculation).
```

Also s/HMAC/MAC/ in step 17.

Reply:
stupid question, but why are the key IDs to be verified different from the device key?

Reply:
> stupid question, but why are the key IDs to be verified different from the device key?

The key IDs to be verified may include the cross-signing keys